### PR TITLE
Test that user from an excluded group cannot share resource

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Add acceptance test to assert user from a shareapi excluded group is not able to share files and folders

## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by jasson99** [comment]
@dpakach @skshetry @HariBhandari07

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 1]
```diff
@@ -174,6 +175,25 @@ Given(
   }
 )
 
+Given('the administrator has enabled shareapi exclude groups', function () {
+  return runOcc(
+    [
+      'config:app:set core shareapi_exclude_groups --value=yes'
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/sharingContext.js]: Does not this require to delete the set configs on `After` hooks? If it does, then, this PR needs to wait for #2269 to be merged.

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 2]
```diff
@@ -162,54 +162,28 @@ Feature: Sharing files and folders with internal users
...
-    And the administrator has browsed to the admin sharing settings page
-    And user "user3" has created folder "/common"
-    And user "user3" has shared folder "/common" with user "user1"
-    And the administrator has enabled exclude groups from sharing from the admin sharing settings page
-    When the administrator adds group "grp1" to the group sharing blacklist using the webUI
-    Then user "user1" should not be able to share folder "/common" with user "User Four" using the sharing API
+    And user "user3" has shared file "testimage.jpg" with user "user1"
+    And user "user3" has shared folder "simple-folder" with user "user1"
+    And the administrator has enabled shareapi exclude groups
+    And the administrator has added group "grp1" to the shareapi exclude groups list
```
Comment (skshetry) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
    And the administrator has excluded group "grp1" from sharing
```

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 3]
```diff
@@ -162,54 +162,28 @@ Feature: Sharing files and folders with internal users
...
     And user "user1" has been added to group "grp1"
-    And the administrator has browsed to the admin sharing settings page
-    And user "user3" has created folder "/common"
-    And user "user3" has shared folder "/common" with user "user1"
-    And the administrator has enabled exclude groups from sharing from the admin sharing settings page
-    When the administrator adds group "grp1" to the group sharing blacklist using the webUI
-    Then user "user1" should not be able to share folder "/common" with user "User Four" using the sharing API
+    And user "user3" has shared file "testimage.jpg" with user "user1"
+    And user "user3" has shared folder "simple-folder" with user "user1"
+    And the administrator has enabled shareapi exclude groups
```
Comment (skshetry) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
    And the administrator has enabled exclude groups from sharing
```

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 4]
```diff
@@ -162,54 +162,28 @@ Feature: Sharing files and folders with internal users
...
-    Then user "user1" should not be able to share file "/testimage (2).jpg" with user "User Four" using the sharing API
+    And the administrator has enabled shareapi exclude groups
+    And the administrator has added group "grp1" to the shareapi exclude groups list
+    And user "user1" has logged in using the webUI
+    Then the user should not be able to share file "testimage.jpg" using the webUI
+    And the user should not be able to share folder "simple-folder" using the webUI
 
-  @skip @yetToImplement
-  Scenario: member of a blacklisted from sharing group tries to re-share a folder received as a share
+  Scenario: member of a blacklisted from sharing group tries to re-share a file or folder received as a share
```
Comment (skshetry) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: Please add a test regarding `share`-only scenario, as was previously.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6087/

<p>














</p></details>

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 5]
```diff
@@ -196,6 +197,25 @@ Given(
...
+  )
+})
+
+Given('the administrator has excluded group {string} from sharing', function (group) {
+  return runOcc(
+    [
+      'config:app:set',
+      'core',
+      'shareapi_exclude_groups_list',
+      '--value=["' + group + '"]'
```
Comment (dpakach) [inline tests/acceptance/stepDefinitions/sharingContext.js]: When there is only one group to exclude from sharing then this is fine. But when we want to implement steps like
```
Given the administrator has excluded group "grp1" from sharing
And the administrator has ecluded group "grp2" from sharing
```
This won't work. 
Please make this work for that kind of cases too.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6159/

<p>













</p></details>

**Comment by jasson99** [comment]
please review : @dpakach @HariBhandari07 @skshetry

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6219/

<p>






</p></details>

**Comment by jasson99** [comment]
@skshetry @dpakach @HariBhandari07

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 6]
```diff
@@ -162,84 +163,59 @@ Feature: Sharing files and folders with internal users
...
-    And group "grp1" has been created
-    And user "user1" has been added to group "grp1"
-    And the administrator has browsed to the admin sharing settings page
-    And user "user3" has shared file "/testimage.jpg" with user "user1"
-    And the administrator has enabled exclude groups from sharing from the admin sharing settings page
-    When the administrator adds group "grp1" to the group sharing blacklist using the webUI
-    Then user "user1" should not be able to share file "/testimage (2).jpg" with user "User Four" using the sharing API
+    And the administrator has enabled exclude groups from sharing
+    And the administrator has excluded group "grp1" from sharing
+    And user "user1" has logged in using the webUI
```
Comment (HariBhandari07) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: ``` When user "user1" has logged in using the webUI```

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 7]
```diff
@@ -162,84 +163,59 @@ Feature: Sharing files and folders with internal users
...
-    And user "user3" has shared folder "/common" with user "user1"
-    And the administrator has enabled exclude groups from sharing from the admin sharing settings page
-    When the administrator adds group "grp1" to the group sharing blacklist using the webUI
-    Then user "user1" should not be able to share folder "/common" with user "User Four" using the sharing API
+    And the setting "shareapi_auto_accept_share" of app "core" has been set to "yes"
+    And user "user3" has shared file "testimage.jpg" with user "user1"
+    And user "user3" has shared folder "simple-folder" with user "user1"
+    And the administrator has enabled exclude groups from sharing
+    And the administrator has excluded group "grp1" from sharing
+    And user "user1" has logged in using the webUI
```
Comment (HariBhandari07) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: When step missing

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature (hunk 6 above)
Comment (jasson99) [reply tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: I fixed that

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature (hunk 7 above)
Comment (jasson99) [reply tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: I fixed that

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6257/

<p>


















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6257/

<p>























</p></details>

**Pull request merged**